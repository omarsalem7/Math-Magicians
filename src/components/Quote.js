import React, { useState, useEffect } from 'react';

const Quote = () => {
  const [quote, setQuote] = useState('');
  useEffect(() => {
    fetch('https://api.adviceslip.com/advice')
      .then((response) => response.json())
      .then((data) => setQuote(data.slip.advice));
  }, []);

  return <div>{quote}</div>;
};

export default Quote;
