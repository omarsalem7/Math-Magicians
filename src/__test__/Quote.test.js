import renderer from 'react-test-renderer';
import Quote from '../components/Quote';

describe('Test Quote component', () => {
  it('Quote renders correctly', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
