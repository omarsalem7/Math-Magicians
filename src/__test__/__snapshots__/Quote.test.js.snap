// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Quote component Quote renders correctly 1`] = `
<div>
  
</div>
`;

exports[`Testing Quote component Quote renders correctly 1`] = `
<div>
  
</div>
`;
